scl/cisco/plugin.conf

```diff
@@ -37,25 +37,17 @@
 
 #
 # parses a cisco timestamp with explicit date-parser
-# It ignores msec and year information
 #
 block parser cisco-timestamp-parser(template()) {
     channel {
-        if {
-            # timestamp from Cisco Unified Call Manager, example "Jun 14 11:57:27 PM.685 UTC"
-            # NOTE: drops msecs and timezone
-            filter { match('^[.*]?([A-Za-z]{3} [0-9 ]\d \d{2}:\d{2}:\d{2} ((AM)|(PM)))' template(`template`) flags(store-matches)); };
```

Comment:
These filters are not *read-only* from the message perspective, because of the `store-matches` instruments the filter to save the matched content. Like the `(...)` blocks, and place them into the `$1, $2, $3, ..., $n`.

One way to resolve it is to keep one filter that accepts the same timestamps as either of those three; and also stores the matched timestamp part into `$1`.

Btw the sets of log message it is tested against. (drops a few of these)
```
<189>91: *Aug 27 14:22:52.787: %SYS-5-CONFIG_I: Configured from console by console
<190>61: *Aug 27 11:21:30.639: %SYS-6-LOGGINGHOST_STARTSTOP: Logging to host 10.10.31.78 started - CLI initiated
<189>90: *Aug 27 14:22:09: %SYS-5-CONFIG_I: Configured from console by console
<189>92: *Aug 27 14:23:23 UTC: %SYS-5-CONFIG_I: Configured from console by console
<189>93: *Aug 27 14:23:40.439 UTC: %SYS-5-CONFIG_I: Configured from console by console
<189>97: *Aug 28 2015 02:25:01.199 Buthan: %SYS-5-CONFIG_I: Configured from console by console
<189>98: %SYS-5-CONFIG_I: Configured from console by console
<189>32: 0.0.0.0: *Apr 29 13:59:12.491: %SYS-5-CONFIG_I: Configured from console by console
<189>102: R1: *Aug 28 02:27:33.231 Buthan: %SYS-5-CONFIG_I: Configured from console by console
```

Reply:
Thanks!  So the milliseconds seems to be optional, and "Buthan" as the timezone is rejected as said by @bazsi here:
https://github.com/balabit/syslog-ng/pull/2779#discussion_r294081186

I guess I will have to drop the timezone support for now.